src/main/java/com/hms_networks/americas/sc/thingworx/TWConnectorConsts.java

```diff
@@ -20,6 +20,13 @@
   public static final String CONNECTOR_VERSION =
       TWConnectorConsts.class.getPackage().getImplementationVersion();
 
+  /** Command to ensure the application always runs */
+  public static final String NEXT_RUN =
+      "-heapsize 25M -classpath /usr/flexy-thingworx-connector-"
+          + CONNECTOR_VERSION
+          + "-full.jar -emain "
+          + TWConnectorMain.class.getName();
+
```

Comment:
This is good as is, but I think Java also allows you to do `class.getProtectionDomain().getCodeSource().getLocation().getPath()` to get the jar path. Not sure if that works in Java 1.4 though.

Reply:
I'll give it a go.

Reply:
Throws null pointer 
``` java 
  public static final String NEXT_RUN_CMD_DES =
      "-heapsize 25M -classpath /usr/"+ AZConnectorMain.class.getProtectionDomain().getCodeSource().getLocation().getPath() + " -emain"
          + " com.hms_networks.americas.sc.azure.AZConnectorMain";
```
![image](https://user-images.githubusercontent.com/72526279/153077037-93137606-8bdf-4718-b527-9a12f4d384fa.png)

Reply:
Not surprising....Was worth a shot.